e2e-polybft/e2e/governance_test.go

```diff
@@ -0,0 +1,330 @@
+		require.NoError(t, err)
+
+		proposalDescription := fmt.Sprintf("Change epoch size from %d to %d", oldEpochSize, newEpochSize)
+
+		proposalID := sendProposalTransaction(t, l2Relayer, proposerAcc.Ecdsa,
+			proposalInput, proposalDescription)
+
+		fmt.Println(proposalID)
+
+		currentBlockNumber, err := l2Relayer.Client().Eth().BlockNumber()
+		require.NoError(t, err)
+
+		// wait couple of blocks until proposal delay finishes
+		require.NoError(t, cluster.WaitForBlock(currentBlockNumber+5, time.Minute))
+
+		// vote for the proposal
+		for _, s := range cluster.Servers {
+			voterAcc, err := sidechain.GetAccountFromDir(s.DataDir())
+			require.NoError(t, err)
+
+			sendVoteTransaction(t, proposalID, For, l2Relayer, voterAcc.Ecdsa)
+		}
+
+		// wait for voting period to end
+		require.NoError(t, cluster.WaitForBlock(votingPeriod+15, 3*time.Minute))
+
+		// check if proposal has quorum (if it was accepted)
+		proposalState := getProposalState(t, proposalID, l2Relayer)
+		require.Equal(t, Succeeded, proposalState)
+
+		// queue proposal for execution
+		sendQueueProposalTransaction(t, l2Relayer, proposerAcc.Ecdsa,
+			proposalInput, proposalDescription)
+
+		currentBlockNumber, err = l2Relayer.Client().Eth().BlockNumber()
+		require.NoError(t, err)
+
+		// wait for couple of blocks until execution delay passes
+		require.NoError(t, cluster.WaitForBlock(currentBlockNumber+3, 30*time.Second))
+
+		// execute proposal
+		sendExecuteProposalTransaction(t, l2Relayer, proposerAcc.Ecdsa,
+			proposalInput, proposalDescription)
+
+		currentBlockNumber, err = l2Relayer.Client().Eth().BlockNumber()
+		require.NoError(t, err)
+
+		// check if epoch size changed on NetworkParams
+		networkParamsResponse, err := ABICall(l2Relayer, contractsapi.NetworkParams,
+			ethgo.Address(contracts.NetworkParamsContract), ethgo.ZeroAddress, "epochSize")
+		require.NoError(t, err)
+
+		epochSizeOnNetworkParams, err := types.ParseUint256orHex(&networkParamsResponse)
+		require.NoError(t, err)
+		require.Equal(t, newEpochSize, epochSizeOnNetworkParams.Uint64())
+
+		endOfPreviousEpoch := (currentBlockNumber/oldEpochSize + 1) * oldEpochSize
+		endOfNewEpoch := endOfPreviousEpoch + newEpochSize
+
+		// wait until the new epoch (with new size finishes)
+		require.NoError(t, cluster.WaitForBlock(endOfNewEpoch, 3*time.Minute))
+
+		block, err := l2Relayer.Client().Eth().GetBlockByNumber(
+			ethgo.BlockNumber(endOfPreviousEpoch), false)
+		require.NoError(t, err)
+
+		extra, err := polybft.GetIbftExtra(block.ExtraData)
+		require.NoError(t, err)
+
+		oldEpoch := extra.Checkpoint.EpochNumber
+
+		block, err = l2Relayer.Client().Eth().GetBlockByNumber(
+			ethgo.BlockNumber(endOfNewEpoch), false)
+		require.NoError(t, err)
+
+		extra, err = polybft.GetIbftExtra(block.ExtraData)
+		require.NoError(t, err)
+
+		newEpoch := extra.Checkpoint.EpochNumber
+
+		// check that epochs are sequential
+		require.True(t, newEpoch-oldEpoch == 1)
```

Comment:
Nitpick, I find it slightly more readable:
```suggestion
		require.Equal(t, newEpoch, oldEpoch + 1)
```

Reply:
Done: https://github.com/0xPolygon/polygon-edge/commit/2b42814c7d23f68e9cdcc252c6b71e30e1c2fe76